cartography/intel/aws/ssm.py

```diff
@@ -0,0 +1,193 @@
+    for instance_id in instance_ids:
+        patches = []
+        for page in paginator.paginate(InstanceId=instance_id):
+            patches.extend(page["Patches"])
+        # to avoid complicating the load function, inject the instance ID into the patch
+        for patch in patches:
+            patch["_instance_id"] = instance_id
+        instance_patches.extend(patches)
+    return instance_patches
+
+
+@timeit
+def load_instance_information(
+    neo4j_session: neo4j.Session,
+    data: List[Dict[str, Any]],
+    region: str,
+    current_aws_account_id: str,
+    aws_update_tag: int,
+) -> None:
+    ingest_query = """
+    UNWIND {InstanceInformation} AS instance
+        MERGE (i:SSMInstanceInformation{id: instance.InstanceId})
+        ON CREATE SET i.firstseen = timestamp()
+        SET i.instance_id = instance.InstanceId,
+            i.ping_status = instance.PingStatus,
+            i.last_ping_date_time = instance.LastPingDateTime,
+            i.agent_version = instance.AgentVersion,
+            i.is_latest_version = instance.IsLatestVersion,
+            i.platform_type = instance.PlatformType,
+            i.platform_name = instance.PlatformName,
+            i.platform_version = instance.PlatformVersion,
+            i.activation_id = instance.ActivationId,
+            i.iam_role = instance.IamRole,
+            i.registration_date = instance.RegistrationDate,
+            i.resource_type = instance.ResourceType,
+            i.name = instance.Name,
+            i.ip_address = instance.IPAddress,
+            i.computer_name = instance.ComputerName,
+            i.association_status = instance.AssociationStatus,
+            i.last_association_execution_date = instance.LastAssociationExecutionDate,
+            i.last_successful_association_execution_date = instance.LastSuccessfulAssociationExecutionDate,
+            i.source_id = instance.SourceId,
+            i.source_type = instance.SourceType,
+            i.region = {Region},
+            i.lastupdated = {aws_update_tag}
+        WITH i
+        MATCH (owner:AWSAccount{id: {AWS_ACCOUNT_ID}})
+        MERGE (owner)-[r:RESOURCE]->(i)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {aws_update_tag}
+        WITH i
+        MATCH (owner:AWSAccount{id: {AWS_ACCOUNT_ID}})-[:RESOURCE]->(ec2_instance:EC2Instance{id: i.instance_id})
+        MERGE (ec2_instance)-[r2:HAS_INFORMATION]->(i)
+        ON CREATE SET r2.firstseen = timestamp()
+        SET r2.lastupdated = {aws_update_tag}
+    """
+    for ii in data:
+        ii["LastPingDateTime"] = dict_date_to_epoch(ii, "LastPingDateTime")
+        ii["RegistrationDate"] = dict_date_to_epoch(ii, "RegistrationDate")
+        ii["LastAssociationExecutionDate"] = dict_date_to_epoch(ii, "LastAssociationExecutionDate")
+        ii["LastSuccessfulAssociationExecutionDate"] = dict_date_to_epoch(ii, "LastSuccessfulAssociationExecutionDate")
+
+    neo4j_session.run(
```

Comment:
non-blocker: make this into a transaction.

Reply:
I'm mostly going off examples from other modules. Would it be good to have a PR that goes through and converts all the modules to using transactions? Maybe ideally using a common function that makes it a bit easier. From what I remember, it's a bit of a pain.